feat(2803): Write orbitdb sync data to QSS

islathehut opened the pull request:
### Pull Request Checklist

- [ ] I have linked this PR to a related GitHub issue.
- [ ] I have added a description of the change (and Github issue number, if any) to the root [CHANGELOG.md](https://github.com/TryQuiet/quiet/blob/develop/CHANGELOG.md).

### (Optional) Mobile checklist

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [ ] I have updated base screenshots for visual regression tests

islathehut committed:
Add data sync method

islathehut committed:
Logging/comment updates

islathehut committed:
Merge branch 'develop' into feat/2803-write-orbitdb-data-to-qss

islathehut committed:
Add dead letter queue for sync messages, fix disconnect bug

islathehut committed:
Update changleog

adrastaea commented on packages/backend/src/nest/local-db/local-db.service.ts:
```diff
@@ -344,4 +348,38 @@ export class LocalDbService {
     }
   }
+
+  public async addPendingQssSyncMessage(unsentMessage: QSSDataSyncMessage): Promise<void> {
```
Instead of saving the whole message to the LevelDB, we could just save the array of CIDs which need synced, and avoid duplicating the entire payload on disk.

adrastaea commented on packages/backend/src/nest/local-db/local-db.service.ts:
```diff
@@ -344,4 +348,38 @@ export class LocalDbService {
+    const key = LocalDBKeys.PENDING_QSS_SYNCS
+    const arr: string[] = (await this.get(key)) || []
+    const messageString = JSON.stringify(unsentMessage)
+    if (arr.filter(pendingMessage => pendingMessage === messageString)) {
```
I believe this will always evaluate to true because an empty array in JS is truthy.

adrastaea commented on packages/backend/src/nest/local-db/local-db.service.ts:
```diff
@@ -344,4 +348,38 @@ export class LocalDbService {
+    let arr: string[] = (await this.get(key)) || []
+    for (const message of sentMessages) {
+      const messageString = JSON.stringify(message)
+      arr = arr.filter(pendingMessage => pendingMessage === messageString)
```
```suggestion
      arr = arr.filter(pendingMessage => pendingMessage !== messageString)
```
I think the logic needs to be this to remove messages.

adrastaea commented on packages/backend/src/nest/qss/qss.service.ts:
```diff
@@ -26,29 +27,77 @@ import EventEmitter from 'node:events'
+import { hash } from '../../../../../3rd-party/auth/packages/crypto/dist'
+import { OrbitDbService } from '../storage/orbitDb/orbitDb.service'
+import { LocalDbService } from '../local-db/local-db.service'
+import { TimedQueue } from '../common/timed-queue'
```
```suggestion
```
Unused import

islathehut committed:
Merge branch 'develop' into feat/2803-write-orbitdb-data-to-qss

islathehut committed:
Update packages/backend/src/nest/qss/qss.service.ts

Co-authored-by: Taea <88346289+adrastaea@users.noreply.github.com>

islathehut committed:
Use hashes for pending message sending

- add team ID to orbitdb metadata
- wonky nestjs circular dependency shenanigans
- remove noise key from network info
- add unit tests for pending message db operations
-

islathehut committed:
Update storage.service.spec.ts

islathehut committed:
Merge branch 'develop' into feat/2803-write-orbitdb-data-to-qss

islathehut committed:
Remove updater as a test

islathehut committed:
Merge branch 'develop' into feat/2803-write-orbitdb-data-to-qss

islathehut committed:
Ensure update modal is closed if open

Don't auto install on quit when you haven't chosen to update

Update selectors.ts

Update oneClient.test.ts

islathehut committed:
Update package-lock.json

islathehut committed:
Only stop auto updating on restart in e2e tests

islathehut committed:
Update package.json

islathehut committed:
Update storage.service.spec.ts

islathehut committed:
fix test errors that show up in other tests

islathehut committed:
Merge branch 'develop' into feat/2803-write-orbitdb-data-to-qss

islathehut committed:
Fix test and update types to match latest spec on QSS
